src/mount/bsd.rs

```diff
@@ -371,19 +371,27 @@ impl<'a> Nmount<'a> {
     pub fn nmount(&mut self, flags: MntFlags) -> NmountResult {
         // nmount can return extra error information via a "errmsg" return
         // argument.
+        
+        // N.B. notgull: the last entry here is technically an IoSliceMut.
+        //               this is a workaround to keep soundness in place
+        let mut iovecs: Vec<libc::iovec> =
```

Comment:
Instead of doing a const transmutation, should self.iov be a vec of IoSliceMut instead?

Reply:
I guess the problem is that the C API isn't const-correct.  All of nmount's arguments are _supposed_ to be const, except for the errmsg argument.  So a transmutation like you had is probably Nix's best option.